frontend/src/scenes/saved-insights/savedInsightsLogic.ts

```diff
@@ -120,8 +120,9 @@ export const savedInsightsLogic = kea<savedInsightsLogicType>([
                 }
 
                 // scroll to top if the page changed, except if changed via back/forward
+                const currentScene = sceneLogic.findMounted()?.values
                 if (
-                    router.values.location.pathname === urls.savedInsights() &&
+                    currentScene?.activeScene === Scene.SavedInsights &&
```

Comment:
Fairly sure you can just use:
```suggestion
                if (
                    sceneLogic.values.activeScene === Scene.SavedInsights &&
```
since there's no way for `savedInsightLogic` to be mounted outside of the context of `<AppScene>` – but works either way